src/Compilers/CSharp/Portable/BoundTree/UnboundLambda.cs

```diff
@@ -927,7 +933,7 @@ BoundLambda rebind(BoundLambda lambda)
                     returnType = DelegateReturnTypeWithAnnotations(invokeMethod, out refKind);
                     if (!returnType.HasType || returnType.Type.ContainsTypeParameter())
                     {
-                        var t = (inferredReturnType.HadExpressionlessReturn || inferredReturnType.NumExpressions == 0)
+                        var t = !inferredReturnType.HasReturnValue
```

Comment:
>!inferredReturnType.HasReturnValue [](start = 32, length = 34)

It feels like this refactoring is not semantically equivalent. It looks like before we could have ```HadExpressionlessReturn == true``` and ```inferredReturnType.NumExpressions > 0``` at the same time. The orignal condition would be ```true```. The new condition for the same scenario would be ```false```.